workitem/link/link_repository.go

```diff
@@ -97,6 +97,31 @@ func (r *GormWorkItemLinkRepository) ValidateCorrectSourceAndTargetType(ctx cont
 	return nil
 }
 
+// CheckParentExist returns error if there is an attempt to create more than 1 parent of a workitem.
+func (r *GormWorkItemLinkRepository) CheckParentExist(ctx context.Context, sourceID, targetID uint64, linkTypeID uuid.UUID) error {
```

Comment:
I would suggest changing the signature like this: `IsParentExists(...) (*bool, error)` (No change in input parameters). The existence of parents and other potential errors are different concerns. When calling the function, first check for errors and then for existence value.